src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchShareLogDAO.scala

```diff
@@ -0,0 +1,133 @@
+  *
+  * @param client      The ElasticSearch client
+  * @param indexName   The name of the target share log index in ElasticSearch
+  * @param refreshMode Using IMMEDIATE - the default - is a performance hit but ensures transactionality
+  *                    of updates across multiple users.
+  *                    TODO would NONE be a better option here?
+  */
+class ElasticSearchShareLogDAO(client: TransportClient, indexName: String, refreshMode: RefreshPolicy = RefreshPolicy.IMMEDIATE)
+  extends ShareLogDAO with ElasticSearchDAOSupport with ShareQueries {
+
+  lazy private final val datatype = "sharelog"
+
+  init // checks for the presence of the index
+
+  /**
+    * Logs a record of a user sharing a workspace, group, or method with a user.
+    *
+    * @param userId     The workbench user id
+    * @param sharee     The email of the user being shared with
+    * @param shareType  The type (workspace, group, or method) see `ShareLog`
+    * @return           The record of the share
+    */
+  override def logShare(userId: String, sharee: String, shareType: String): Share = {
+    val share = Share(userId, sharee, shareType, Instant.now)
+    val id = MurmurHash3.stringHash(userId + sharee + shareType).toString
+    val insert = client
+      .prepareIndex(indexName, datatype, id)
+      .setSource(share.toJson.compactPrint, XContentType.JSON)
+      .setCreate(true) // fail the request if the logged share already exists
+      .setRefreshPolicy(refreshMode)
+
+    executeESRequest[IndexRequest, IndexResponse, IndexRequestBuilder](insert)
+    share
+  }
+
+  /**
+    * Gets a share by the ID, a `MurmurHash3` of `userId` + `sharee` + `shareType`
+    *
+    * @param id The ID of the share
+    * @return A record of the share
+    */
+  override def getShare(id: String): Share = {
+    val getSharesQuery = client.prepareGet(indexName, datatype, id)
+    Try(executeESRequest[GetRequest, GetResponse, GetRequestBuilder](getSharesQuery)) match {
+      case Success(get) if get.isExists => get.getSourceAsString.parseJson.convertTo[Share]
+      case Success(_) => throw new FireCloudException(s"share for $id not found")
+      case Failure(f) => throw new FireCloudException(s"error receiving share for $id: ${f.getMessage}")
+    }
+  }
+
+  /**
+    * Gets all shares that have been logged for a workbench user which fall under the
+    * given type of share (workspace, method, group).
+    *
+    * @param userId     The workbench user ID
+    * @param shareType  The type (workspace, group, or method) - if left blank returns all shares
+    * @return A list of `ShareLog.Share`s
+    */
+  override def getShares(userId: String, shareType: Option[String] = None): Seq[Share] = {
+    val getSharesRequest = client
+      .prepareSearch(indexName)
+      .setQuery(
+        if (shareType.isDefined)
+          userSharesOfType(userId, shareType.get)
+        else
+          userShares(userId)
+      )
+      // possibly sort and limit size
```

Comment:
if you don't set a page size, you get the default of 10 - which means that after an end user shares a bunch of times, we won't be returning all sharees.